modules/camera/camera_win.cpp

```diff
@@ -30,65 +30,632 @@
+		temp.is_rgb24 = (video_format == MFVideoFormat_RGB24);
+
+		// Set format string
+		FeedFormat format;
+		if (video_format == MFVideoFormat_RGB24) {
+			format.format = "RGB24";
+		} else if (video_format == MFVideoFormat_RGB32) {
+			format.format = "RGB32";
+		} else if (video_format == MFVideoFormat_ARGB32) {
+			format.format = "ARGB32";
+		} else if (video_format == MFVideoFormat_NV12) {
+			format.format = "NV12";
+		} else if (video_format == MFVideoFormat_YUY2) {
+			format.format = "YUY2";
+		} else if (video_format == MFVideoFormat_UYVY) {
+			format.format = "UYVY";
+		} else if (video_format == MFVideoFormat_YVYU) {
+			format.format = "YVYU";
+		} else if (video_format == MFVideoFormat_I420) {
+			format.format = "I420";
+		} else if (video_format == MFVideoFormat_IYUV) {
+			format.format = "IYUV";
+		} else if (video_format == MFVideoFormat_YV12) {
+			format.format = "YV12";
+		} else if (video_format == MFVideoFormat_MJPG) {
+			format.format = "MJPG";
+		}
+		format.width = width;
+		format.height = height;
+		format.frame_numerator = numerator;
+		format.frame_denominator = denominator;
+
+		temp.format = format;
+		temp_formats.append(temp);
+
+		imf_media_type->Release();
+	}
+
+	// Second pass: add formats prioritizing RGB24
+	for (const TempFormat &temp : temp_formats) {
+		FormatKey key;
+		key.width = temp.format.width;
+		key.height = temp.format.height;
+		key.frame_numerator = temp.format.frame_numerator;
+		key.frame_denominator = temp.format.frame_denominator;
+		key.is_rgb24 = temp.is_rgb24;
+
+		// Check if this combination already exists
+		bool should_add = true;
+		for (const FormatKey &seen : seen_formats) {
+			if (seen == key) {
+				// If we already have this combination and it was RGB24, skip non-RGB24
+				if (seen.is_rgb24 && !temp.is_rgb24) {
+					should_add = false;
+					break;
+				}
+				// If we have non-RGB24 and this is RGB24, we'll replace it
+				else if (!seen.is_rgb24 && temp.is_rgb24) {
+					// Remove the existing non-RGB24 format
+					for (int j = formats.size() - 1; j >= 0; j--) {
+						if (formats[j].width == key.width &&
+								formats[j].height == key.height &&
+								formats[j].frame_numerator == key.frame_numerator &&
+								formats[j].frame_denominator == key.frame_denominator) {
+							formats.remove_at(j);
+							format_guids.remove_at(j);
+							format_mediatypes.remove_at(j);
+							break;
+						}
+					}
+					// Update seen_formats
+					for (int j = 0; j < seen_formats.size(); j++) {
+						if (seen_formats[j] == key) {
+							seen_formats.write[j].is_rgb24 = true;
+							break;
+						}
+					}
+				}
+				// Both are the same type, skip
+				else {
+					should_add = false;
+					break;
+				}
+			}
+		}
+
+		if (should_add) {
+			// Add to seen formats
+			seen_formats.append(key);
+
+			// Add format
+			this->formats.append(temp.format);
+			this->format_guids.append(temp.video_format);
+			this->format_mediatypes.append(temp.media_type_index);
```

Comment:
We generally don't use `this->` in Godot, unless absolutely needed.

```suggestion
			formats.append(temp.format);
			format_guids.append(temp.video_format);
			format_mediatypes.append(temp.media_type_index);
```